### DISCUSSION TITLE
Changable `TSDataset` attributes

### DISCUSSION THREAD

**Discussion by d-a-bunin** [etna-team/etna#422] [Improvements]
### Description

Currently, in `TSDataset` we store public attributes:
- `df`
- `raw_df`
- `df_exog`
- `known_future`
- `freq`
- `hierarchical_structure`
- `current_df_level`
- `current_df_exog_level`

It make these attributes look available for writing after the init, but in most cases changing them after init could break the consistency of the class.

We should somehow prohibit changing them in most cases and make them changable only in cases when we are sure about the consistency.

### Thoughts

We could wrap these attributes into `property`.

### Related

- #418
- #433

**Comment by egoriyaa** [comment]
Solved in #630
